persist/fs/retriever.go

```diff
@@ -285,6 +308,23 @@ func (r *blockRetriever) Stream(
 	startTime time.Time,
 	onRetrieve block.OnRetrieveBlock,
 ) (xio.SegmentReader, error) {
+	// This should never happen
+	if len(r.seekerMgrs) < 1 {
```

Comment:
It might be cleaner to assign a single seeker mgr to be the "main" one for bloom filter lookups and use that one here?  We can also assign a no-op one to it (like a mock seeker mgr) by default so that if we haven't opened the block retriever or w/e it just returns similarly to a "not found".

Reply:
I don't like the idea that if you trying and Stream() before Open() you get the equivalent of a "not found", I think that should probably be an "use before open" error or something. The existing code kind of captures that, although I can make one of the seeker managers the "main" one. Should I just add another field to the struct for that?